packages/scene-composer/src/layouts/SceneLayout/components/FoldableContainer.tsx

```diff
@@ -1,64 +1,41 @@
-import React, { useState } from 'react';
-import styled from 'styled-components';
+import React, { useEffect, useState } from 'react';
 import { Icon, IconProps } from '@awsui/components-react';
-import * as awsui from '@awsui/design-tokens';
 
 import { Direction } from './utils';
-
-const WrapperFoldLeft = styled.div`
-  display: flex;
-  flex-direction: row;
-  align-items: stretch;
-  height: 100%;
-`;
-
-const WrapperFoldRight = styled.div`
-  display: flex;
-  flex-direction: row-reverse;
-  align-items: stretch;
-  height: 100%;
-`;
-
-const Content = styled.div`
-  flex: 1;
-`;
-
-const Handle = styled.div`
-  flex: none;
-  display: flex;
-  flex-direction: column;
-  justify-content: center;
-  align-items: center;
-  border-left: 1px solid ${awsui.colorBorderDividerDefault};
-  border-right: 1px solid ${awsui.colorBorderDividerDefault};
-
-  &:hover {
-    background-color: ${awsui.colorBackgroundDropdownItemHover};
-  }
-`;
+import './FoldableContainer.scss';
 
 type FoldableContainerProps = React.PropsWithChildren<{
   direction: Direction;
+  open: boolean;
+  setIsOpen: (boolean) => void;
+  ref: React.ReactFragment | undefined;
 }>;
 
-const FoldableContainer: React.FC<FoldableContainerProps> = ({ direction, children }: FoldableContainerProps) => {
-  const [fold, setFold] = useState(false);
+const FoldableContainer: React.FC<FoldableContainerProps> = ({
+  direction,
+  children,
+  open,
+  setIsOpen,
+}: FoldableContainerProps) => {
+  const [fold, setFold] = useState(!open);
+  const isLeft = direction === 'Left';
+  const isRight = direction === 'Right';
```

Comment:
Nit: A small improvement could be to define `isRight` as `!isLeft`, assuming there will never be more than two directions.